src/node.cpp

```diff
@@ -103,10 +105,23 @@ void publish_scan(ros::Publisher *pub,
     pub->publish(scan_msg);
 }
 
-bool checkRPLIDARHealth(RPlidarDriver * drv)
+void publish_diag(const std::string &serial_port, int8_t level, const std::string &message, ros::Publisher &diag_pub) 
+{
+    diagnostic_msgs::DiagnosticArray darray;
+    diagnostic_msgs::DiagnosticStatus status;
+    status.level = level;
+    status.name = "rplidar driver";
+    status.hardware_id = serial_port;
+    status.message = message;
+    darray.status.push_back(status);
+    diag_pub.publish(darray);
```

Comment:
so you are saying some diagnostics are only published once at startup, and are otherwise redundant with the `scan` publish freq diagnostic? It looks from this that such a diagnostic is not very helpful